src/Bundle/Controller/Admin/ThemeCatalogController.php

```diff
@@ -0,0 +1,113 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0  Academic Free License (AFL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\Mbo\Bundle\Controller\Admin;
+
+use PrestaShop\Module\Mbo\Adapter\ExternalContentProvider;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\ServiceUnavailableHttpException;
+use Symfony\Component\Templating\EngineInterface;
+
+/**
+ * Responsible of "Improve > Design > Themes Catalog" page display.
+ */
+class ThemeCatalogController extends FrameworkBundleAdminController
+{
+    /**
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request)
+    {
+        $response = new Response();
+
+        try {
+            $externalContentProvider = new ExternalContentProvider();
+
+            $content = $this->getTemplateEngine()->render(
+                '@Modules/ps_mbo/views/templates/admin/controllers/theme_catalog/addons_store.html.twig',
+                [
+                    'pageContent' => $externalContentProvider->getContent($this->getAddonsUrl($request)),
+                    'layoutHeaderToolbarBtn' => [],
+                    'layoutTitle' => $this->trans('Themes Catalog', 'Admin.Navigation.Menu'),
+                    'requireAddonsSearch' => true,
+                    'requireBulkActions' => false,
+                    'showContentHeader' => true,
+                    'enableSidebar' => true,
+                    'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+                    'requireFilterStatus' => false,
+                    'level' => $this->authorizationLevel($request->attributes->get('_legacy_controller')),
+                ]
+            );
+            $response->setContent($content);
+        } catch (ServiceUnavailableHttpException $exception) {
```

Comment:
you'd better use an exception listener for that :)

Reply:
I see your point, but I need a more specific exception for that.
Exception listener will be subscribed to event kernel.exception, maybe it's overkill to catch only one exception from a module ?

Reply:
ok :)